util/chplenv/chpl_atomics.py

```diff
@@ -11,12 +11,17 @@
 def get(flag='target'):
     if flag == 'network':
         atomics_val = overrides.get('CHPL_NETWORK_ATOMICS')
+        comm_val = chpl_comm.get()
         if not atomics_val:
-            comm_val = chpl_comm.get()
             if comm_val in ['ofi', 'ugni'] and get('target') != 'locks':
                 atomics_val = comm_val
             else:
                 atomics_val = 'none'
+        elif atomics_val == 'gasnet':
+            error("CHPL_NETWORK_ATOMICS=gasnet not supported")
+        elif atomics_val != 'none' and atomics_val != comm_val:
+            error("CHPL_NETWORK_ATOMICS=%s incompatible with CHPL_COMM=%s" %
+                  (atomics_val, comm_val))
```

Comment:
I think you should add an `else: error("unsupported value for CHPL_NETWORK_ATOMICS")`, in case a user has something like `export CHPL_NETWORK_ATOMICS=blah`